conn.go

```diff
@@ -178,6 +174,14 @@ var (
 	errBadWriteOpCode      = errors.New("websocket: bad write message type")
 	errWriteClosed         = errors.New("websocket: write closed")
 	errInvalidControlFrame = errors.New("websocket: invalid control frame")
+
+	// ErrCloseSent is returned when the application writes a message to the
+	// connection after sending a close message.
+	ErrCloseSent = errors.New("websocket: close sent")
```

Comment:
Move ErrCloseSent and ErrReadLimit back to where they were. There's are a lot of changes to review. It's easier without changes unrelated to the feature.